gemrb/plugins/SDLVideo/SDL20Video.cpp

```diff
@@ -551,7 +551,7 @@ Holder<Sprite2D> SDL20VideoDriver::GetScreenshot(Region r, const VideoBufferPtr&
 
 	SDL_Texture* target = SDL_GetRenderTarget(renderer);
 	if (buf) {
-		auto texture = static_cast<SDLTextureVideoBuffer*>(drawingBuffer)->GetTexture();
```

Comment:
what makes you believe that `drawingBuffer` and `but` are interchangeable? even if sometimes true is it always? it would seem not or else you would have the same problem. In short I dont under stand how this fixes anything.

Reply:
Even if they're not (which i highly doubt, because as i checked, screenshots and savefile screenshots are working as before), it's a bad idea to check for buf, but then cast to drawingBuffer. 
This function already has everything it needs as passed arguments, but doesn't know anything about the state the drawingBuffer is in. We *can* instead put a drawingBuffer check (and then we don't need buf argument at all), but then instead of crashes we will get to the SDL_SetRenderTarget(renderer, nullptr), which, i dunno, will make an entirely black screenshot maybe, or whatever it does.

Reply:
yup, seems like a mistake now that I look at it